ouroboros-consensus-test/src/Test/Util/ChainDB.hs

```diff
@@ -0,0 +1,111 @@
+                     configSecurityParam)
+import           Ouroboros.Consensus.Fragment.InFuture (CheckInFuture (..))
+import qualified Ouroboros.Consensus.Fragment.Validated as VF
+import           Ouroboros.Consensus.HardFork.History.EraParams (EraParams,
+                     eraEpochSize)
+import           Ouroboros.Consensus.Ledger.Basics (LedgerConfig)
+import           Ouroboros.Consensus.Ledger.Extended (ExtLedgerState)
+import           Ouroboros.Consensus.Protocol.Abstract
+import           Ouroboros.Consensus.Storage.ChainDB hiding
+                     (TraceFollowerEvent (..))
+import           Ouroboros.Consensus.Storage.FS.API (SomeHasFS (..))
+import qualified Ouroboros.Consensus.Storage.ImmutableDB as ImmutableDB
+import           Ouroboros.Consensus.Storage.ImmutableDB.Chunks.Internal
+                     (simpleChunkInfo)
+import qualified Ouroboros.Consensus.Storage.LedgerDB.DiskPolicy as LedgerDB
+import qualified Ouroboros.Consensus.Storage.VolatileDB as VolatileDB
+import           Ouroboros.Consensus.Util.IOLike hiding (invariant)
+import           Ouroboros.Consensus.Util.ResourceRegistry (ResourceRegistry)
+
+import qualified Test.Util.FS.Sim.MockFS as Mock
+import           Test.Util.FS.Sim.STM (simHasFS)
+
+import           Ouroboros.Consensus.Block.Abstract
+import           Test.Util.FS.Sim.MockFS
+
+
+-- | A vector with an element for each database of a node
+--
+-- The @db@ type parameter is instantiated by this module at types for mock
+-- filesystems; either the 'MockFS' type or reference cells thereof.
+data NodeDBs db = NodeDBs
+  { nodeDBsImm :: db
+  , nodeDBsVol :: db
+  , nodeDBsLgr :: db
+  } deriving (Functor)
+
+emptyNodeDBs :: MonadSTM m => m (NodeDBs (StrictTVar m MockFS))
+emptyNodeDBs = NodeDBs
+  <$> uncheckedNewTVarM Mock.empty
+  <*> uncheckedNewTVarM Mock.empty
+  <*> uncheckedNewTVarM Mock.empty
+
+-- | Minimal set of arguments for creating a ChainDB instance for testing purposes.
+data MinimalChainDbArgs m blk = MinimalChainDbArgs {
+    mcdbTopLevelConfig :: TopLevelConfig blk
+  , mcdbChunkInfo      :: ImmutableDB.ChunkInfo
+  -- ^ Specifies the layout of the ImmutableDB on disk.
+  , mcdbInitLedger     :: ExtLedgerState blk
+  -- ^ The initial ledger state.
+  , mcdbRegistry       :: ResourceRegistry m
+  -- ^ Keeps track of non-lexically scoped resources.
+  , mcdbNodeDBs        :: NodeDBs (StrictTVar m MockFS)
+  -- ^ File systems underlying the immutable, volatile and ledger databases.
+  -- Would be useful to default this to StrictTVar's containing empty MockFS's.
+  }
+
+-- | Utility function to get a default chunk info in case we have EraParams available.
+mkTestChunkInfo ::  LedgerConfig blk ~ EraParams => TopLevelConfig blk -> ImmutableDB.ChunkInfo
+mkTestChunkInfo = simpleChunkInfo . eraEpochSize . topLevelConfigLedger
+
+-- | Creates a default set of of arguments for ChainDB tests.
+fromMinimalChainDbArgs :: (MonadThrow m, MonadSTM m, ConsensusProtocol (BlockProtocol blk))
```

Comment:
Maybe place the constraints below so that we can avoid long lines. See the style guide for an example on how multi-line constraints are written.